app/src/main/java/com/hyphenate/chatuidemo/call/FloatWindow.java

```diff
@@ -0,0 +1,316 @@
+    }
+
+    private void unregisterBroadcast() {
+        if (localBroadcastManager != null && broadcastReceiver != null) {
+            localBroadcastManager.unregisterReceiver(broadcastReceiver);
+        }
+    }
+
+    Handler handler = new Handler() {
+        @Override public void handleMessage(Message msg) {
+            //super.handleMessage(msg);
+            switch (msg.what) {
+                case REFRESH_UI:
+                    Intent intent = (Intent) msg.obj;
+                    EMCallStateChangeListener.CallState callState =
+                            (EMCallStateChangeListener.CallState) intent.getExtras().get("call_state");
+                    EMCallStateChangeListener.CallError callError =
+                            (EMCallStateChangeListener.CallError) intent.getExtras().get("call_error");
+                    refreshCallView(callState, callError);
+                    break;
+                case REFRESH_TIME:
+                    refreshCallTime();
+                    break;
+            }
+        }
+    };
+
+    /**
+     * refresh call ui
+     */
+    private void refreshCallView(EMCallStateChangeListener.CallState callState, EMCallStateChangeListener.CallError callError) {
+        switch (callState) {
+            case CONNECTING:
+                break;
+            case CONNECTED:
+                break;
+            case ACCEPTED:
+                break;
+            case DISCONNECTED:
+                CallManager.getInstance().removeFloatWindow();
+                break;
+            case NETWORK_DISCONNECTED:
+                Toast.makeText(context, "Remote network disconnected!", Toast.LENGTH_SHORT).show();
+                break;
+            case NETWORK_NORMAL:
+                Toast.makeText(context, "Remote network connected!", Toast.LENGTH_SHORT).show();
+                break;
+            case NETWORK_UNSTABLE:
+                if (callError == EMCallStateChangeListener.CallError.ERROR_NO_DATA) {
+                    Toast.makeText(context, "No call data!", Toast.LENGTH_SHORT).show();
+                } else {
+                    Toast.makeText(context, "Remote network unstable!", Toast.LENGTH_SHORT).show();
+                }
+                break;
+            case VIDEO_PAUSE:
+                Toast.makeText(context, "Remote pause video", Toast.LENGTH_SHORT).show();
+                break;
+            case VIDEO_RESUME:
+                Toast.makeText(context, "Remote resume video", Toast.LENGTH_SHORT).show();
+                break;
+            case VOICE_PAUSE:
+                Toast.makeText(context, "Remote pause voice", Toast.LENGTH_SHORT).show();
+                break;
+            case VOICE_RESUME:
+                Toast.makeText(context, "Remote resume voice", Toast.LENGTH_SHORT).show();
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void refreshCallTime() {
```

Comment:
there is similar method in class DateUtils, please consider to reuse them or put this method to that class.